husky_components/datagrid/decorators/tiles-view.js

```diff
@@ -0,0 +1,226 @@
+            this.sandbox = this.datagrid.sandbox;
+
+            // merge defaults with options
+            this.options = this.sandbox.util.extend(true, {}, defaults, options);
+            this.setVariables();
+        },
+
+        /**
+         * Sets the variables used by this decorator to their initial values
+         */
+        setVariables: function() {
+            this.rendered = false;
+            this.$el = null;
+            this.$tiles = {};
+        },
+
+        /**
+         * Method to render this view.
+         *
+         * @param data object containing the data which is rendered
+         * @param $container dom-element to render datagrid in
+         */
+        render: function(data, $container) {
+            this.renderContainer($container);
+            data.embedded.map(this.renderRecord.bind(this));
+            if (!!this.options.showAddNewTile) {
+                this.renderAddNewTile();
+            }
+            this.rendered = true;
+        },
+
+        /**
+         * Renders the main container of the decorator
+         *
+         * @param {Object} $container The element to render the decorator into.
+         */
+        renderContainer: function($container) {
+            this.$el = $('<div class="tiles-view"/>');
+            $container.append(this.$el);
+        },
+
+        /**
+         * Renders the add new tile and binds its click event
+         */
+        renderAddNewTile: function() {
+            var $addNew = $(_.template(templates.addNewTile, {
+                icon: this.options.addNewIcon,
+                title: this.sandbox.translate(this.options.translations.addNew)
+            }));
+
+            $addNew.on('click', function() {
+                this.sandbox.emit(ADD_CLICKED.call(this));
+            }.bind(this));
+
+            this.$el.append($addNew);
+        },
+
+        /**
+         * Renders a single record into the view
+         *
+         * @param {Object} record The data of the record
+         */
+        renderRecord: function(record) {
+            record = processContentFilters.call(this, record);
+            var title = this.getTitleString(record),
+                description = this.getDescriptionSring(record);
+            this.$tiles[record.id] = $(_.template(templates.tile, {
+                icon: this.options.icon,
+                title: title,
+                croppedTitle: this.sandbox.util.cropMiddle(this.getTitleString(record), 20),
+                description: this.sandbox.util.cropMiddle(description, 20)
+            }));
+
+            this.bindTileDomEvents(record.id);
+            this.$el.append(this.$tiles[record.id]);
+        },
+
+        /**
+         * Binds dom events for a single tile
+         *
+         * @param {Object} tileId The id of the tile to bind the events for
+         */
+        bindTileDomEvents: function(tileId) {
+            this.$tiles[tileId].on('click', function() {
```

Comment:
Wouldn't it be easier and more efficient to bind this event once with a filter, instead for every single tile?